src/Mvc/test/Microsoft.AspNetCore.Mvc.Core.Test/Routing/ActionEndpointDataSourceBaseTest.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Patterns;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Primitives;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Routing
+{
+    // There are some basic sanity tests here for the details of how actions
+    // are turned into endpoints. See ActionEndpointFactoryTest for detailed tests.
+    public abstract class ActionEndpointDataSourceBaseTest
+    {
+        [Fact]
+        public void Endpoints_AccessParameters_InitializedFromProvider()
+        {
+            // Arrange
+            var actions = new Mock<IActionDescriptorCollectionProvider>();
+            actions.Setup(m => m.ActionDescriptors).Returns(new ActionDescriptorCollection(new List<ActionDescriptor>
+            {
+                CreateActionDescriptor(new { Name = "Value", }, "/Template!"),
+            }, 0));
+
+            var dataSource = CreateDataSource(actions.Object);
+
+            // Act
+            var endpoints = dataSource.Endpoints;
+
+            // Assert
+            var endpoint = Assert.IsType<RouteEndpoint>(Assert.Single(endpoints));
+            Assert.Equal("Value", endpoint.RoutePattern.RequiredValues["Name"]);
+            Assert.Equal("/Template!", endpoint.RoutePattern.RawText);
+        }
+
+        
+
```

Comment:
nit: so liney

Reply:
You should add this to an editorconfig and automate it